cloudformation/application-setup-basic/stacks/application/update-application-stack.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+export AWS_PAGER=""
+
+DOCKER_IMAGE_URL=$1
+USER_POOL_CLIENT_SECRET=$2
+
+# upload the stack files
+aws s3 cp ./ s3://aws101.dev/stacks/application --recursive
+
+# create the parent stack with all child stacks
+aws cloudformation update-stack \
+  --stack-name aws101-application-parent \
+  --template-body file://application.yml \
+  --parameters \
+    ParameterKey=NetworkStackTemplateUrl,ParameterValue=https://s3.amazonaws.com/aws101.dev/stacks/application/network.yml \
+    ParameterKey=ServiceStackTemplateUrl,ParameterValue=https://s3.amazonaws.com/aws101.dev/stacks/application/service.yml \
+    ParameterKey=ServiceStackImageUrl,ParameterValue=$DOCKER_IMAGE_URL \
+    ParameterKey=ServiceStackUserPoolClientSecret,ParameterValue=$USER_POOL_CLIENT_SECRET \
+    ParameterKey=RegistryStackName,ParameterValue=aws101-container-registry 2> update_error
+
+# Sadly, the AWS CLI returns an error when no update is to be performed. But we want to
+# call it a success if the stack is up-to-date. So, we have to work around this by
+# catching and grepping the error output. See https://github.com/aws/aws-cli/issues/3625.
```

Comment:
This was annoying. There's a lot of duplication in the bash scripts because we're doing this special handling for every stack now. 

I believe we should look at CDK at some point to see if can do the same with reusable code...